### ISSUE TITLE
V4: Inconsistent className styles merging

### ISSUE THREAD

**Issue by ShaneKeney** [marklawlor/nativewind#665]
**Describe the bug**
Merging styles doesn't seem to be consistent.

e.g. 

```
const Swatch = ({ className, children }: { className?: string; children: React.ReactNode }) => {
  return (
    <View
      className={`flex h-32 w-full justify-between bg-slate-500 bg-red-500 p-[16px]`}
    >
      {children}
    </View>
  );
};
```

specifically:

```
      className={`flex h-32 w-full justify-between bg-slate-500 bg-red-500 p-[16px]`}
```

The above `className` is showing as background color: `slate-500` instead of `red-500`.
Removing `bg-slate-500` shows as `bg-red-500` as expected.
Adding `bg-slate-500` after `bg-red-500` shows slate as expected.

If I change to 

```
      className={`flex h-32 w-full justify-between bg-slate-500 bg-slate-900 p-[16px]`}

```

It shows `bg-slate-900` as expected.  

So there seems to be an inconsistency in how the styles are being applied when overriding or duplicate styles are passed to the `className` 

**To Reproduce**
Can't seem to reproduce on an Expo Snack.  But any development client build I use suffers from the issue.

**Expected behavior**
Based on docs https://www.nativewind.dev/v4/guides/custom-components the last utility style class should be the one that shows.

**Expo Snack**
Can't seem to reproduce on an Expo Snack.  But any development client build I use suffers from the issue.

**Comment by marklawlor** [comment]
This is expected behaviour, the docs for custom components are yet to be updated. I have created new docs on how classes are merged here https://www.nativewind.dev/v4/core-concepts/style-specificity (this is still a WIP, let me know if it can be improved).

TLDR: NativeWindv4 aligns more with Tailwind CSS in how styles are applied. `bg-slate-500` has a higher specificity than `bg-red-500` (due to rule order), so it will be picked.

If you wish to change this behaviour, you can use tools like https://www.npmjs.com/package/tailwind-merge which will modify the classname

**Comment by ShaneKeney** [comment]
@marklawlor  Got it.  Thanks!  Based on the symptoms, I figured that was what was going on.  I just wasn't sure if that was meant to be intended behavior.  

Learning Tailwind through Nativewind on React Native so thanks for the help.
